merge master(parent-> improvements) to mongo-rdkit

vvrubel opened the pull request:
Этот pr будет о том как смержила ветку mongo-rdkit и improvements
poetry.lock опять за свое ~1,122

vvrubel committed:
WIP: операции над базой

vvrubel committed:
Merge remote-tracking branch 'origin/master' into mongo-rdkit

vvrubel committed:
smiles search endpoint

vvrubel committed:
UPD: имя файла

vvrubel committed:
search file

vvrubel committed:
WIP: операции с базой данных

vvrubel committed:
WIP: операции с базой данных

vvrubel committed:
WIP: операции с базой данных

vvrubel committed:
Merge branch 'master' into merge_rdkit_w_improvements

churnikov commented on .env.sample:
```diff
@@ -3,8 +3,12 @@ MONGO_HOST=""
 MONGO_DB_NAME=""
 MONGO_DB_COLLECTION=""
+MONGO_SCHEMA_COLLECTION=""
+MONGO_TEST_COLLECTION=""
+MONGO_TEST_SCHEMA_COLLECTION=""
```
А
Что?
Почему так?

Если хочешь другие значения для тестирования использовать, то просто другой .env файл используй

vvrubel replied:
> А
> Что?
> Почему так?
> 
> Если хочешь другие значения для тестирования использовать, то просто другой .env файл используй

а есть где посмотреть, как settinngs.py примерно будет выглядеть с двумя .env файлами?

churnikov replied:
Ты можешь менять в тестах на нужный .env файл, как описано в пункте 2
https://pydantic-docs.helpmanual.io/usage/settings/#dotenv-env-support

churnikov commented on README.md:
```diff
@@ -196,8 +192,23 @@ Options:
 управление зависимостями этих библиотек с использованием `poetry`.  
 Если вы хотите использовать окружение конды и вам хочется узнать, как установить `conda` рядом с 
-`pyenv` наиболее безболезненно – читайте [тут](https://confluence.biocad.ru/x/vi1QCw).
+`pyenv` наиболее безболезненно – читайте [тут](https://confluence.biocad.ru/x/vi1QCw).  
```
```suggestion
`pyenv` наиболее безболезненно – читайте [тут](https://confluence.biocad.ru/x/vi1QCw).
```

vvrubel replied:
> Suggested change 
> `pyenv` наиболее безболезненно – читайте [тут](https://confluence.biocad.ru/x/vi1QCw).  
> `pyenv` наиболее безболезненно – читайте [тут](https://confluence.biocad.ru/x/vi1QCw).

ты пробел убрал?

churnikov replied:
Да
Это видно по этой фиговине
Удалено 2 пробела
<img width="92" alt="image" src="https://user-images.githubusercontent.com/8545082/125610742-be430fa1-5f7b-44b4-b4ad-81fb49fa0140.png">

churnikov commented on molecad/data/console/cli.py:
```diff
@@ -56,46 +56,17 @@ def cli():
-
-
-@cli.command(
+@pubchem_to_db.command(
```
Ох уж эти мелкие рефакторинги

vvrubel replied:
> Ох уж эти мелкие рефакторинги

каюсь...

churnikov commented on molecad/data/db.py:
```diff
@@ -1,50 +1,124 @@
+    q_mol = Chem.MolFromSmiles(sub_smiles)
+    substructure.AddPatternFingerprints(rdkit_schema)
+    index_lst = substructure.SubSearch(q_mol, rdkit_schema, chirality=False)
+    # index_lst = substructure.SubSearchNaive(q_mol, rdkit_schema, chirality=False)
```
.

churnikov replied:
Ну ты поняла, да?)

vvrubel replied:
> Ну ты поняла, да?)

ды

churnikov commented on molecad/data/db.py:
```diff
@@ -1,50 +1,124 @@
+    return list(cursor)
+
+
+def main():
+    prepare_for_search()
+    res = search("CC(=O)OC")
+    print([i['CanonicalSMILES'] for i in res])
+
+
+if __name__ == "__main__":
+    main()
```
Тянет на тест кейс
Но забей сейчас

churnikov commented on molecad/data/downloader.py:
```diff
@@ -56,32 +56,25 @@ def url_builder(
-    ``build_url``; по умолчанию - ``Out.JSON``.
-    .. note:: В текущей версии сервиса получение ответа от сервера возможно только в формате JSON.
-    :return: URL-адрес, который используется для запроса в базу данных Pubchem.
+    :param ids: Последовательность из значений одинакового типа, которые интерпретируются как
```
Почему так сильно поменялось
Тут есть что-то важное?

vvrubel replied:
> Почему так сильно поменялось
> Тут есть что-то важное?

было много функций, которые делали одно и тоже, и затрудняли понимание кода на мой взгляд, я их сжала в одну. Только почему-то git не показывает удаленные кроме докстрингов у меня..

churnikov replied:
Понятно

churnikov commented on molecad/main.py:
```diff
@@ -0,0 +1,29 @@
+    Volume3D: Optional[Union[float, int]]
+
+
+@app.get("/v1/compound", response_model=List[Molecule])
+def get_compounds(smiles: str, skip: int, limit: int) -> List[Molecule]:
+    return search(smiles)[skip:][:limit]
```
Если эта ручка ещё не актуальна, то убери этот код вообще отсюда

vvrubel replied:
> Если эта ручка ещё не актуальна, то убери этот код вообще отсюда

ок

churnikov commented on molecad/data/downloader.py:
```diff
@@ -163,8 +156,9 @@ def execute_requests(start: int, stop: int, maxsize: int = 100) -> Iterator[Dict
-    :return: итератор по записям из базы данных Pubchem - 'Compound'.
+    :return: Генератор запросов к базе данных Pubchem - 'Compound'.
     """
+    #TODO
```
А что за TODO?

vvrubel replied:
> А что за TODO?

тоже забыла убрать заметку

vvrubel reviewed (commented):
скоро будет коммит

vvrubel committed:
NEW: структура проекта, отдельные конфигурационные файлы для dev/prod окружения

vvrubel committed:
UPD: решены ошибки при инициализации настроек,
NEW: теперь при загрузке данных в базу с помощью команды `populate`, сразу генерируются предварительные настройки базы для api

vvrubel committed:
WIP: агрегация для поиска

churnikov commented on README.md:
```diff
@@ -68,38 +68,47 @@ MONGO_HOST|Хост базы данных MongoDB; по умолчанию -> `"
 
 Options:
-  --help  Show this message and exit.
+  --setup [PROD|DEV]  Опция, позволяющая выбирать конфигурационный файл,
```
```suggestion
  --env [PROD|DEV]  Опция, позволяющая выбирать конфигурационный файл,
```

Принято так это называть

churnikov commented on molecad/cli.py:
```diff
@@ -14,19 +24,36 @@
+    "сохранения полученной информации в файлы формата JSON. Также с помощью утилиты можно "
+    "загружать полученные данные в базу MongoDB и подготавливать их для поиска молекул."
+    "Для выполнения команд, которые устанавливают соединение с базой данных, необходимо "
+    "указать опцию --prod/--dev для выбора рабочей базы.",
```
Ошибка

churnikov commented on molecad/validator.py:
```diff
@@ -14,65 +15,72 @@
+        return all(isinstance(tag, PropertyTags) for tag in tags)
+
+
+def check_smiles(smiles) -> None:
```
Тип

churnikov reviewed (approved):
Можешь забить на коменты, если что
Время поджимает

vvrubel committed:
NEW: запросы в базу

vvrubel committed:
Revert "NEW: запросы в базу"

This reverts commit a4edac88ae3c95c7c03278809c6c88a2229cd1a7.
